scipy/signal/_splinemodule.c

```diff
@@ -18,166 +18,6 @@ convert_strides(npy_intp* instrides,npy_intp* convstrides,int size,int N)
   }
 }
 
-
-static char doc_cspline2d[] = "out = cspline2d(input, lambda=0.0, precision=-1.0)\n"
-"\n"
-"    Coefficients for 2-D cubic (3rd order) B-spline.\n"
-"\n"
-"    Return the third-order B-spline coefficients over a regularly spaced\n"
-"    input grid for the two-dimensional input image.\n"
-"\n"
-"    Parameters\n"
-"    ----------\n"
-"    input : ndarray\n"
-"        The input signal.\n"
-"    lambda : float\n"
-"        Specifies the amount of smoothing in the transfer function.\n"
-"    precision : float\n"
-"        Specifies the precision for computing the infinite sum needed to apply mirror-\n"
-"        symmetric boundary conditions.\n"
-"\n"
-"    Returns\n"
-"    -------\n"
-"    output : ndarray\n"
-"        The filtered signal.\n"
-"\n"
-"    Examples\n"
-"    --------\n"
-"    Examples are given :ref:`in the tutorial <tutorial-signal-bsplines>`.\n"
-"\n";
-
-
-static PyObject *cspline2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
-{
-  PyObject *image=NULL;
-  PyArrayObject *a_image=NULL, *ck=NULL;
-  double lambda = 0.0;
-  double precision = -1.0;
-  int thetype, M, N, retval=0;
-  npy_intp outstrides[2], instrides[2];
-
-  if (!PyArg_ParseTuple(args, "O|dd", &image, &lambda, &precision)) return NULL;
-
-  thetype = PyArray_ObjectType(image, NPY_FLOAT);
-  thetype = PyArray_MIN(thetype, NPY_DOUBLE);
-  a_image = (PyArrayObject *)PyArray_FromObject(image, thetype, 2, 2);
-  if (a_image == NULL) goto fail;
-
-  ck = (PyArrayObject *)PyArray_SimpleNew(2, PyArray_DIMS(a_image), thetype);
-  if (ck == NULL) goto fail;
-  M = PyArray_DIMS(a_image)[0];
-  N = PyArray_DIMS(a_image)[1];
-
-  convert_strides(PyArray_STRIDES(a_image), instrides, PyArray_ITEMSIZE(a_image), 2);
-  outstrides[0] = N;
-  outstrides[1] = 1;
-
-  if (thetype == NPY_FLOAT) {
-    if ((precision <= 0.0) || (precision > 1.0)) precision = 1e-3;
-    retval = S_cubic_spline2D((float *)PyArray_DATA(a_image),
-                              (float *)PyArray_DATA(ck),
-                              M, N, lambda, instrides, outstrides, precision);
-  }
-  else if (thetype == NPY_DOUBLE) {
-    if ((precision <= 0.0) || (precision > 1.0)) precision = 1e-6;
-    retval = D_cubic_spline2D((double *)PyArray_DATA(a_image),
-                              (double *)PyArray_DATA(ck),
-                              M, N, lambda, instrides, outstrides, precision);
-  }
-
-  if (retval == -3) PYERR("Precision too high.  Error did not converge.");
-  if (retval < 0) PYERR("Problem occurred inside routine");
```

Comment:
Could you comment why is this no longer possible or needed?

Reply:
This is no longer needed since the function does not call malloc anymore